assembl/static/js/app/views/groups/groupContent.js

```diff
@@ -22,46 +22,124 @@ var groupContent = Marionette.CompositeView.extend({
   childViewContainer: ".groupBody",
   childView: PanelWrapper,
   panel_borders_size: 1,
+  minPanelSize:AssemblPanel.prototype.minimized_size,
+  criticalSize:600,
 
   initialize: function(options) {
     var that = this;
+    this.options = options;
     this.collection = this.model.get('panels');
-    this.groupContainer = options['groupContainer'];
+    this.groupContainer = this.options['groupContainer'];
     setTimeout(function() {
       if (!that.isViewDestroyed()) {
         var navView = that.findViewByType(PanelSpecTypes.NAV_SIDEBAR);
         if (navView) {
           navView.setViewByName(that.model.get('navigationState'), null);
         }
       }
+      that.resizePanel(true);
     }, 200); //FIXME:  Magic delay...
+    $(window).on("resize",function(){
+      that.resizePanel(true);
+    });
   },
   events: {
     'click .js_closeGroup': 'closeGroup'
   },
-  collectionEvents: {
-    'add remove reset change': 'containerAdjustGridSize'
+  resizePanel:function(skipAnimation){
+    var that = this;
+    var screenSize = window.innerWidth;
+    var animationDuration = 1000;
+    _.each(that.groupContainer.collection.models,function(group){
+      _.each(group.attributes.panels.models,function(elm){
+        var panelMinWidth = elm.get('minWidth');
+        var isPanelMinimized = elm.get('minimized');
+        var panelWidth = that.getPanelWidth(panelMinWidth,isPanelMinimized);
+        var panelId = '#' + elm.cid;
+        var panel = $(that.groupContainer.el).find(panelId);
+        if(skipAnimation){
+          panel.css({'min-width':panelMinWidth});
+          panel.width(panelWidth);
+        }else{
+          var totalWidth = that.getTotalWidth();
+          if(totalWidth < screenSize){
+            panel.css({'min-width':0});
+            panel.animate({'width': panelWidth}, animationDuration, 'swing',function(){
+              panel.css({'min-width':panelMinWidth});
+            });
+          }else{
+            if(screenSize <= that.criticalSize){
+              panel.animate({'min-width': panelMinWidth}, animationDuration, 'swing')
+            }else{
+              panel.css({'min-width':0});
+              panel.animate({'width': panelMinWidth}, animationDuration, 'swing',function(){
+                panel.css({'min-width':panelMinWidth});
+              });
+            }
+          }
+        }
+      });
+    });
+  },
+  getPanelWidth:function(panelMinWidth,isPanelMinimized){
+    var screenSize = window.innerWidth;
+    var panelWIdth = 0;
+    if(isPanelMinimized){
+      panelWIdth = this.minPanelSize;
+    }else{
+      if(screenSize > this.criticalSize){
+        var totalWidth = this.getTotalWidth();
+        var panelWidthInPercent = (panelMinWidth * 100) / totalWidth;
```

Comment:
Braaaaaaaaavo :) !!!
Le 12 mai 2016 6:13 AM, "Marc-Antoine Parent" <notifications@github.com> a
écrit :

> In assembl/static/js/app/views/groups/groupContent.js
> <https://github.com/ImaginationForPeople/assembl/pull/34#discussion_r62996205>
> :
>
> > +              });
> > +            }
> > +          }
> > +        }
> > +      });
> > +    });
> > +  },
> > +  getPanelWidth:function(panelMinWidth,isPanelMinimized){
> > +    var screenSize = window.innerWidth;
> > +    var panelWIdth = 0;
> > +    if(isPanelMinimized){
> > +      panelWIdth = this.minPanelSize;
> > +    }else{
> > +      if(screenSize > this.criticalSize){
> > +        var totalWidth = this.getTotalWidth();
> > +        var panelWidthInPercent = (panelMinWidth * 100) / totalWidth;
>
> Ah... pas ce que j'avais en tête, mais je crois que tu as pris une bonne
> décision au total. Les gridsize se voulaient donner des proportions plus
> harmonieuses mais n'y sont jamais arrivées; mais se battaient avec les
> minima en pratique. Ton code donne des animations plus fluides, et je crois
> que c'est ce qui compte, bravo. Je suis surtout impressionné que tu aies pu
> faire l'économie des corrections pour les panneaux minimisés et obtenir un
> bon résultat.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/ImaginationForPeople/assembl/pull/34/files/6becb51695dcbb475c0d8e56ea28f139f57e3330#r62996205>
>